### DISCUSSION TITLE
testing sqlite

### DISCUSSION THREAD

**Discussion by bpolasek519** [pytest-dev/pytest#10674] [Q&A]
Hi!

I'm a new dev and I am building a web app and using sqlite to cache my data. My boss (rightfully) got irritated with me yesterday that I wasn't using tests to test the CRUD operations. I haven't mainly because I'm not sure how. I have used pytest on other parts of the project, but haven't jumped in with databases yet. Can anyone direct me (or help me) to a very straightforward way of how to use pytest with databases?

I'm also not using SQLAlchemy which a lot of tutorials I have seen have. Not sure if it's that important or if I can do without. Thanks!

**Comment by nicoddemus** [comment]
Hi,

Googling for `testing sqlite pytest` yields this as the first result:

https://smirnov-am.github.io/pytest-testing_database/

Did not read thoroughly, but it does seem up-to-date at a glance.

Hope this helps!

**Comment by arthuRHD** [comment]
Hi, before going anywhere can you explain a little bit more how you do these CRUD operations ? Is it through an API ?

And what you want to do are integration tests. Are you sure this the right kind of test ?
I did these errors in the past, that's why I want to be sure.

Whether or not you need to test the data, you should test the real data by testing the HTTP calls. There's numerous tools for that, check out [Postman](https://learning.postman.com/docs/getting-started/introduction/). Otherwise, just [mock](https://docs.python.org/3/library/unittest.mock.html) method calls that interact with sqlite inside your unit tests.

@bpolasek519 keep me updated 📫
